### DISCUSSION TITLE
Support extracted labels in template variables (RE or PATTERN)

### DISCUSSION THREAD

**Discussion by wedwardbeck** [grafana/grafana#41117] [Dashboards, Panels & Visualizations]
**What would you like to be added:**
Use of PATTERN in template variable queries for datasource Loki. 

**Why is this needed:**
Using PATTERN to reduce labels being sent in on logs to Loki, it's easy to filter logs and produce the same panels/queries as with normal labels, and the parsing work on the client side is reduced (thanks).  However, this is not usable in template variables that I can find, and [forum responses](https://community.grafana.com/t/variable-based-on-field-value-json/43301) seem to confirm this.  

While regex can be used to get close, in some cases this still fails.  
Example query using PATTERN:
```{log="invoice_extractor"}| pattern `<_>"log":"<level>  <logtime> - <original_pdf>, <pages> pages, <message>"<_> ` ```
![image](https://user-images.githubusercontent.com/19333237/139481631-c0bd56f1-e599-4f48-a748-2242891bc56d.png)

Extracted labels:
![image](https://user-images.githubusercontent.com/19333237/139481708-1591d0b3-4e93-45aa-9a8c-16f806f04dc8.png)

Desired template variable use (sample - guessing at syntax):
```
label_values({host="faxserver",job="tail",log="invoice_extractor"}|pattern `<_>"log":"<level> `,level)
```
![image](https://user-images.githubusercontent.com/19333237/139482248-25776983-d372-40bc-9a2a-b533bc61f5f2.png)

**Comment by shivtej-opsverse** [comment]
@wedwardbeck did you find a workaround this?
I am trying to do the same for a label extracted using regular expression.

**Reply by wedwardbeck** [reply]
No sorry I ended up changing plans since I could not get it working at that time.  I need to revisit sometime to see what new options there are.

**Comment by VytenisSabTelia** [comment]
Same here. I need Template variables based on Detected fields

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
